### ISSUE TITLE
Break Foundational headers into their own libary.

### ISSUE THREAD

**Issue by Sqeaky** [BlackToppStudios/Mezzanine#110]
The most core headers like "datatypes.h" are used by most BTS software yet are included only in the mezzanine. These need to be separated into a separate piece of software and somehow managed. That management is to be reserved for another issue.

**Comment by Sqeaky** [comment]
This depends on #113 and #114

**Comment by Sqeaky** [comment]
StaticFoundation Broken out on its Jagati package and that is half of this.
